xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java

```diff
@@ -17,49 +17,67 @@
 package io.grpc.xds;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static io.grpc.ConnectivityState.IDLE;
+import static io.grpc.ConnectivityState.SHUTDOWN;
+import static io.grpc.internal.ServiceConfigUtil.getBalancerPolicyNameFromLoadBalancingConfig;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
 import io.grpc.Attributes;
 import io.grpc.ConnectivityStateInfo;
 import io.grpc.EquivalentAddressGroup;
 import io.grpc.LoadBalancer;
 import io.grpc.LoadBalancerRegistry;
 import io.grpc.Status;
 import io.grpc.internal.ServiceConfigUtil;
+import io.grpc.xds.XdsLbState.SubchannelStore;
 import io.grpc.xds.XdsLbState.XdsComms;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
-import javax.annotation.CheckReturnValue;
+import java.util.concurrent.atomic.AtomicReference;
 import javax.annotation.Nullable;
 
 /**
  * A {@link LoadBalancer} that uses the XDS protocol.
  */
 final class XdsLoadBalancer extends LoadBalancer {
 
-  final Helper helper;
+  private static final Attributes.Key<AtomicReference<ConnectivityStateInfo>> STATE_INFO =
+      Attributes.Key.create("io.grpc.xds.XdsLoadBalancer.stateInfo");
+
+  private static final ImmutableMap<String, Object> DEFAULT_FALLBACK_POLICY =
+      ImmutableMap.of("round_robin", (Object) ImmutableMap.<String, Object>of());
+
+  private final SubchannelStore subchannelStore = new SubchannelStore();
+  private final Helper helper;
+  private final FallbackManager fallbackManager;
 
   @Nullable
   private XdsLbState xdsLbState;
 
+  private Map<String, Object> fallbackPolicy;
+
   XdsLoadBalancer(Helper helper) {
```

Comment:
It's better to inject `LoadBalancerRegistry` through the constructor, so that tests can construct `LoadBalancerRegistry` with fake child policies and will not collide with the ones from default registry. The XDS balancer factory will pass the default registry.

ditto for the registry in `FallbackManager`.